tsdb/head.go

```diff
@@ -516,6 +524,29 @@ func (h *Head) loadWAL(r *wal.Reader, multiRef map[uint64]uint64, mmappedChunks
 		}(inputs[i], outputs[i])
 	}
 
+	wg.Add(1)
+	exemplarsInput = make(chan record.RefExemplar, 300)
+	go func(input <-chan record.RefExemplar) {
+		for e := range input {
+			ms := h.series.getByID(e.Ref)
+			if ms == nil {
+				unknownRefs.Inc()
+				continue
+			}
+
+			if e.T < h.minValidTime.Load() {
+				continue
+			}
+			// At the moment the only possible error here is out of order exemplars, which we shouldn't see when
+			// replaying the WAL, so lets just log the error if it's not that type.
+			err = h.exemplars.AddExemplar(ms.lset, exemplar.Exemplar{Ts: e.T, Value: e.V, Labels: e.Labels})
+			if err != nil && err == storage.ErrOutOfOrderExemplar {
```

Comment:
Did you mean this? (Also should we check for duplicate error too?)
```suggestion
			if err != nil && err != storage.ErrOutOfOrderExemplar {
```

Reply:
`exemplarStorage.AddExemplar` will never actually return the duplicate error, since we don't really treat it as an error (duplicates are expected, just not ingested).

Reply:
`Did you mean this?` also yes